Assignment1/obsAvoidance/obsAvoidance.ino

```diff
@@ -0,0 +1,34 @@
+#include <Smartcar.h>
+
+const int forwardSpeed   = 70; 
+const int stoppingSpeed = 0; 
+const int stopDistance = 60;
+
+ArduinoRuntime arduinoRuntime;
+
+BrushedMotor leftMotor(arduinoRuntime, smartcarlib::pins::v2::leftMotorPins);
+BrushedMotor rightMotor(arduinoRuntime, smartcarlib::pins::v2::rightMotorPins);
+DifferentialControl control(leftMotor, rightMotor);
+
+SimpleCar car(control);
+
+const int TRIGGER_PIN           = 6; // D6
+const int ECHO_PIN              = 7; // D7
+const unsigned int MAX_DISTANCE = 100;
+
+SR04 frontSensor(arduinoRuntime, TRIGGER_PIN, ECHO_PIN, MAX_DISTANCE);
+
+void setup()
+{
+    Serial.begin(9600);
+    car.setSpeed(forwardSpeed);
+}
+
+void loop()
+{ 
+     if (frontSensor.getDistance() < stopDistance && frontSensor.getDistance() > 0){
```

Comment:
What happens if the distance is diferent in the two different calls you're making?

Reply:
We think that it may not provide an accurate stopping distance in this case? We think a solution would be to have the distance monitored constantly instead. Thanks for the feedback!

Reply:
No, that's not what I meant. If the first call to `getDistance` returns `0` (i.e. no obstacle) and the second call returns `200` because an object far away is suddenly detected, then the car would stop before it should.

Reply:
n that case, would it be better to have a variable that is assigned the frontSensor.getDistance() returned value?
```suggestion
    int distance = frontSensor.getDistance();
     if (distance < stopDistance && distance > 0){
```

Reply:
👍

Reply:
Thank you :smiley: